### ISSUE TITLE
Is it possible to not have both `self` & `related` in links

### ISSUE THREAD

**Issue by simonv3** [fotinakis/jsonapi-serializers#12]
As per the spec, [you don't need both](http://jsonapi.org/format/#document-resource-object-relationships), and I feel like it's unnecessary and unwieldy to do both in my routes.

Any thoughts on accomplishing this?

**Comment by fotinakis** [comment]
If you override the serializer method `self_link`, `relationship_self_link`, or `relationship_related_link` and return `nil`, it will not be included.

See the methods here: https://github.com/fotinakis/jsonapi-serializers#more-customizations

**Comment by simonv3** [comment]
That's what I had tried, and came to an intermediate solution with. When I set:

```
  def relationship_self_link(attribute_name)
    nil
  end
```

I get:

```
relationships": {
    "pictures": {
        "links": {
            "self": null,
            "related": "/api/v1/crops/551e3e4853696d1d44030000/pictures"
        },
        "data": [...]
    }
}
```
Note the `self: null`. Same goes for the `relationship_related_link` method. The `self_link` method does indeed not get included. Is that what's meant to happen?

**Comment by fotinakis** [comment]
Ah actually it looks like I only implemented that for the root object and not the relationships, ie this line doesn't have a conditional:
https://github.com/fotinakis/jsonapi-serializers/blob/master/lib/jsonapi-serializers/serializer.rb#L91

So, not possible at the moment, but should be pretty easy to add, let me see if I can do this easily now.

**Comment by fotinakis** [comment]
Ok, fixed and released as v0.2.4. Now if you override relationship_self_link or relationship_related_link to return nil, the link will be excluded from the serialized object. This also fixes #6 because if both return `nil`, the full `links` object will be excluded.

Thanks for the suggestion, let me know if that doesn't work for you.

**Comment by simonv3** [comment]
Awesome, thanks! Just updated the gem and it looks like it's fixed.

Thanks so much for the awesomely quick fix!
